tools/serve/serve.py

```diff
@@ -657,17 +661,15 @@ def start(config, ssl_environment, routes, **kwargs):
     paths = {"doc_root": config["doc_root"],
              "ws_doc_root": config["ws_doc_root"]}
 
-    external_config = normalise_config(config, ports)
-
-    ssl_config = get_ssl_config(config, external_config["domains"].values(), ssl_environment)
+    ssl_config = get_ssl_config(config, config["domains"].values(), ssl_environment)
```

Comment:
Somehow CI is failing with this throwing `KeyError: 'domains'`. I'm confused as to how that's possible, but I haven't dug into it.

Reply:
Well, there are more than one code paths to serve.start(), all of which need to be changed if we now want to make start() take a normalized config.

The main function of serve.py is only one of the paths. Another path I know is via wptrunner, in particular `wptrunner/environment.py` when entering TestEnvironment. https://github.com/w3c/web-platform-tests/blob/0e5650daa71f750cdbdee6546b0b56b01c24128b/tools/wptrunner/wptrunner/environment.py#L107

P.S. A few lines above, still in start method, there's a `domains = get_subdomains(host)` which I think is no longer used. (GitHub wouldn't let me comment on just a few lines from the patch...)